.github/workflows/build.yml

```diff
@@ -74,63 +73,33 @@ jobs:
             build_type: Release
             code_coverage: false
             static: false
-    uses: ./.github/workflows/build_impl.yml
-    with:
-      runs_on: ${{ matrix.os }}
-      container: ${{ matrix.container }}
-      conan_profile: ${{ matrix.conan_profile }}
-      build_type: ${{ matrix.build_type }}
-      code_coverage: ${{ matrix.code_coverage }}
-      static: ${{ matrix.static }}
-      unit_tests: true
-      integration_tests: true
-      clio_server: true
-
-  test:
-    name: Run Tests
-    needs: build
-    strategy:
-      fail-fast: false
-      matrix:
-        include:
```

Comment:
I think I fixed a bug here, because this matrix only has 4 scenarios, but we're building for 5

Reply:
No, tests for gcc debug run in build stage.

Reply:
Ah, got it, thanks, as part of code coverage step.

I think there is no harm to run it with and without coverage.
And I will soon change the place coverage runs after this PR is merged, so let's keep it as is for now.

Reply:
What do you mean with and without coverage?
Tests compiled by gcc in Debug are always with coverage enabled. Also it takes 40 seconds to run them in debug, so it would be better to avoid running them twice.

Reply:
We shouldn't build and run tests with coverage twice, but it's ok to fix it in a separate PR.